core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -116,8 +116,10 @@ def _load_endpoints(self) -> None:
             self.configuration["endpoints"] = []
         endpoints = self.configuration["endpoints"]
         for endpoint in endpoints:
-            if not self.mavlink_manager.add_endpoint(Endpoint(**endpoint)):
-                warn(f"Could not load endpoint {endpoint}")
+            try:
+                self.mavlink_manager.add_endpoint(Endpoint(**endpoint))
```

Comment:
Is there no way to us to identify via pylint which function can raise ?
Why there return type is not being used now ? Should we don't warn when a endpoint already exists in the list ?

Reply:
I solved a problem there but created another one.

The problem was that, when the Endpoint was not valid, it raised a NotImplementedError, which was not being catched, so I have put it inside a try/except block. The problem is that we still have the situation of duplicated endpoints, which was catched before and is not anymore.

Will fix that.

Reply:
Still, I don't think that returning false is very explicitly about what is happening.

Let's say that I don't know how the logic of the endpoints are just a list appending. Maybe it was a database. How do I know, just by receiving a false response, what happened? Was that false a database store failing? Was a duplicated endpoint? Was a validation error? The validation error is the only one were I truly know what happened, because it raised an error, instead of a status.

Reply:
That's why I think raising errors and catching them explicitly on the proper interface is the correct way to do it:

```python
def add_endpoint(self, endpoint):
    if duplicated: raise ValueError("Duplicated endpoint")
    if not_validated: raise ValueError("Endpoint not valid for current router")
    if ...: raise ...
    self._endpoints.append(endpoint)
```

Reply:
To me, the clear solution is to use a different architecture to fix the issue:
https://pypi.org/project/result/

Reply:
But we can do that in a future PR, this is already getting long. 
Just make sure to create issues for all relevant points here.

Reply:
I *really* liked the idea of the result library. I will open an issue for that and we can discuss about turning it a standard for the repo.